azurerm/resource_arm_storage_account.go

```diff
@@ -909,6 +918,11 @@ func expandStorageAccountNetworkRules(d *schema.ResourceData) *storage.NetworkRu
 	networkRule := networkRules[0].(map[string]interface{})
 	networkRuleSet := &storage.NetworkRuleSet{}
 
+	if networkRule["default_action"].(string) == "Allow" {
```

Comment:
@katbyte Forgive me because I dont have much knowledge regarding enums in golang. I think you are asking if we can use enums so that we dont have to do string comparison as enum comparison (int) is much faster. Correct me if I am wrong, from what I gather we dont have a way to specify enums.

https://github.com/Azure/azure-sdk-for-go/blob/master/services/storage/mgmt/2018-02-01/storage/models.go#L118

Reply:
what i mean is convert the constant to a string and use that instead of hard coding the value as you have, enum was probably not the best terminology to use!

Reply:
Understood. I believe you were looking at old commit. I did get same feedback from Tom and [it was updated to constant](
https://github.com/terraform-providers/terraform-provider-azurerm/pull/3255/commits/15d95645b0b675618fbc5dddcfe1a34c7caa33a2).